### PR TITLE
[test][integration] Start the multikueue test clusters in parallel.

### PR DESCRIPTION
**Pull request by trasc**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

[test][integration] Start the multikueue test clusters in parallel.

#### Which issue(s) this PR fixes:

Relates to #2993

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: test/integration/multikueue/suite_test.go [hunk 1]
```diff
@@ -233,10 +234,34 @@ var _ = ginkgo.BeforeSuite(func() {
 		managerFeatureGates = []string{"JobManagedBy=true"}
 	}
 
-	// pass nil setup since the manager for the manage cluster is different in some specs.
-	managerTestCluster = createCluster(nil, managerFeatureGates...)
-	worker1TestCluster = createCluster(managerSetup)
-	worker2TestCluster = createCluster(managerSetup)
+	ginkgo.By("creating the clusters", func() {
+		mtx := sync.Mutex{}
```
Comment (mimowo) [inline test/integration/multikueue/suite_test.go:238]: What is the mtx needed for? I see it guards assignment to `managerTestCluster`, `worker1TestCluster`, `worker2TestCluster`, but these variables don't seem used concurrently.

**Review by mimowo — COMMENTED** [review]
Please also add to the description what is the time gain from this on CI.

File: test/integration/multikueue/suite_test.go (hunk 1 above)
Comment (trasc) [reply test/integration/multikueue/suite_test.go]: Good catch, it was left over form a previous try.

**Review by mimowo — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**